osc-server/src/main/java/org/osc/core/broker/service/securitygroup/ListSecurityGroupBindingsBySgService.java

```diff
@@ -62,16 +63,19 @@
         Set<VirtualSystem> vsSet = sg.getVirtualizationConnector().getVirtualSystems();
         long order = -1;
 
-        //Existing bindings
+        // Existing bindings
         for (SecurityGroupInterface sgInterface : sg.getSecurityGroupInterfaces()) {
             VirtualSystem vs = sgInterface.getVirtualSystem();
             vsSet.remove(vs);
-            VirtualSystemPolicyBindingDto virtualSystemBindingDto = new VirtualSystemPolicyBindingDto(vs.getId(), vs
-                    .getDistributedAppliance().getName(), sgInterface.getMgrPolicy() == null ? null : sgInterface.getMgrPolicy().getId(),
-                            FailurePolicyType.valueOf(sgInterface.getFailurePolicyType().name()),
-                            sgInterface.getOrder());
+			Set<Long> policyIds = sgInterface.getMgrPolicies().stream().map(Policy::getId).collect(Collectors.toSet());
+            VirtualSystemPolicyBindingDto virtualSystemBindingDto = new VirtualSystemPolicyBindingDto(vs.getId(),
+                    vs.getDistributedAppliance().getName(), policyIds,
+                    FailurePolicyType.valueOf(sgInterface.getFailurePolicyType().name()), sgInterface.getOrder());
             virtualSystemBindingDto.setMarkedForDeletion(sgInterface.getMarkedForDeletion());
             virtualSystemBindingDto.setBinded(true);
+			if (this.apiFactoryService.supportsMultiplePolicies(vs)) {
```

Comment:
can be reduced to virtualSystemBindingDto.setMultiplePoliciesSupported(this.apiFactoryService.supportsMultiplePolicies(vs))